tendermint/src/public_key.rs

```diff
@@ -39,6 +50,44 @@ pub enum PublicKey {
     Secp256k1(Secp256k1),
 }
 
+impl DomainType<RawPubKeyResponse> for PublicKey {}
+
+impl TryFrom<RawPubKeyResponse> for PublicKey {
+    type Error = Error;
+
+    // This does not check if the underlying pub_key_ed25519 has the right size.
+    // The caller needs to make sure that this is actually the case.
+    fn try_from(value: RawPubKeyResponse) -> Result<Self, Self::Error> {
+        match &value
+            .pub_key
+            .ok_or_else(|| format_err!(error::Kind::InvalidKey, "empty pubkey"))?
+            .sum
+            .ok_or_else(|| format_err!(error::Kind::InvalidKey, "empty sum"))?
```

Comment:
Can we introduce a let binding here to help with readability a little bit?

Reply:
I tried, check in the next commit